tests/drivers/watchdog/wdt_error_cases/src/main.c

```diff
@@ -0,0 +1,1001 @@
+#define WDT_FEED_CAN_STALL
+
+/* Align tests to the specific target: */
+#if defined(CONFIG_SOC_NRF54L15)
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0x07CFFFFFU)
+#undef WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#undef WDT_DISABLE_IS_NOT_SUPPORTED
+#undef WDT_FEED_CAN_STALL
+#else
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0xFFFFFFFFU)
+#endif
+
+static const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
+static struct wdt_timeout_cfg m_cfg_wdt0;
+
+/* Following variables are incremented in WDT callbacks
+ * to indicate whether interrupt was fired or not.
+ */
+volatile uint32_t m_test_06b_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_06B_TAG (0x12345678U)
+volatile uint32_t m_test_08b_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08B_TAG (0x23456789U)
+volatile uint32_t m_test_08d_A_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08D_A_TAG (0x3456789AU)
+volatile uint32_t m_test_08d_B_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08D_B_TAG (0x456789ABU)
+
+static void wdt_test_06b_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_06b_value = TEST_06B_TAG;
+}
+
+static void wdt_test_08b_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08b_value = TEST_08B_TAG;
+}
+
+static void wdt_test_08d_A_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08d_A_value = TEST_08D_A_TAG;
+}
+
+static void wdt_test_08d_B_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08d_B_value = TEST_08D_B_TAG;
+}
+
+#if !defined(WDT_DISABLE_IS_NOT_SUPPORTED)
+/**
+ * @brief wdt_disable() negative test
+ *
+ * Confirm that wdt_disable() returns
+ * -EFAULT when watchdog instance is not enabled.
+ *
+ */
+ZTEST(wdt_coverage, test_01_wdt_disable_before_wdt_setup)
+{
+	int ret;
+
+	ret = device_is_ready(wdt);
+	zassert_true(ret, "WDT device is not ready, got unexpected value of %d", ret);
+
+	/* Call wdt_disable before enabling wdt */
+	ret = wdt_disable(wdt);
+	zassert_true(ret == -EFAULT,
+		     "Calling wdt_disable before watchdog was started should return -EFAULT (-14), "
+		     "got unexpected value of %d",
+		     ret);
+}
+#else
+ZTEST(wdt_coverage, test_01_wdt_disable_before_wdt_setup)
+{
+	ztest_test_skip();
+}
+#endif
+
+/**
+ * @brief wdt_setup() negative test
+ *
+ * Confirm that wdt_setup() returns error value or ASSERTION FAIL
+ * when it's called before wdt_install_timeouts().
+ *
+ */
+ZTEST(wdt_coverage, test_02_wdt_setup_before_setting_timeouts)
+{
+	int ret;
+
+	ret = device_is_ready(wdt);
+	zassert_true(ret, "WDT device is not ready, got unexpected value of %d", ret);
```

Comment:
This check should be moved to a `ztest_suite_before_t` function, so that it does not have to be added to every single test case. See [here](https://docs.zephyrproject.org/latest/develop/test/ztest.html#test-fixtures) for the documentation.

Reply:
Done.